File: src/main/java/club/xiaojiawei/config/LogListenerConfig.java
package club.xiaojiawei.config;

import club.xiaojiawei.listener.log.AbstractLogListener;
import club.xiaojiawei.listener.log.DeckLogListener;
import club.xiaojiawei.listener.log.PowerLogListener;
import club.xiaojiawei.listener.log.ScreenLogListener;
import jakarta.annotation.Resource;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

/**
 * LogListener的责任链配置
 * @author 肖嘉威
 * @date 2023/7/5 14:48
 */
@Configuration
public class LogListenerConfig {

    @Resource
    private DeckLogListener deckLogListener;
    @Resource
    private PowerLogListener powerLogListener;
    @Resource
    private ScreenLogListener screenLogListener;

    @Bean
    public AbstractLogListener logListener(){
        deckLogListener.setNextLogListener(powerLogListener).setNextLogListener(screenLogListener);
        return deckLogListener;
    }
}

File: Hearthstone-Script/src/main/java/club/xiaojiawei/hsscript/config/LogListenerConfig.kt
package club.xiaojiawei.hsscript.config

import club.xiaojiawei.hsscript.listener.log.AbstractLogListener
import club.xiaojiawei.hsscript.listener.log.DeckLogListener
import club.xiaojiawei.hsscript.listener.log.PowerLogListener
import club.xiaojiawei.hsscript.listener.log.ScreenLogListener

/**
 * @author 肖嘉威
 * @date 2024/10/14 10:00
 */
object LogListenerConfig {

    val logListener: AbstractLogListener = DeckLogListener

    init {
        DeckLogListener
            .setNextLogListener(PowerLogListener)
            .setNextLogListener(ScreenLogListener)
    }

}